src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      authContext: {
+        authData: {},
+        isAuthenticated: this.isAuthenticated,
+        setCurrentUser: this.setCurrentUser,
+        setCurrentUserProfile: this.setCurrentUserProfile
+      }
+    };
+  }
+
+  componentDidMount = () => {
+    // Sync auth session stored in the localStorage with Context
+    // TBH, I'm not sure if we even need context
+    this.setCurrentUser(AuthService.getSessionAuthData());
+  };
+
+  isAuthenticated = () => {
+    const {
+      userId,
+      expiresAt,
+      auth0AccessToken,
+      graphcoolToken
+    } = this.state.authContext.authData;
+    return (
+      Date.now() < expiresAt && userId && auth0AccessToken && graphcoolToken
+    );
+  };
+
+  setCurrentUser = authData => {
+    this.setState({
+      authContext: {
+        ...this.state.authContext,
+        authData
+      }
+    });
+  };
+
+  setCurrentUserProfile = userProfile => {
+    AuthService.setAuthProfileSession(userProfile); // I don't really like this side-effect being here
+    this.setState({
+      authContext: {
+        ...this.state.authContext,
+        authData: {
+          ...this.state.authContext.authData,
+          userProfile
+        }
+      }
+    });
+  };
+
```

Comment:
Let's also add a listener to `storage` changes somewhere (probably in the `AuthProvider`) to update the state/handle logouts if the storage changes in another tab. See current code for this [here](https://github.com/codefordenver/members/blob/ee4580e3a66b83c2ced90cca0ce95433111b6b4f/src/utils/withAuthSession.js#L69-L74)